lib/wasi.js

```diff
@@ -118,14 +114,30 @@ class WASI {
     this[kInstance] = undefined;
   }
 
+  setupInstance(instance, memory) {
+    validateObject(instance, 'instance');
+    validateObject(instance.exports, 'instance.exports');
```

Comment:
We should also move the `if (this[kStarted]) throw new ERR_WASI_ALREADY_STARTED()` error here as we should't allow this to be called twice.